### PR TITLE
update docs for __virtual__ and __virtualname__

### PR DESCRIPTION
**Pull request by jacobhammons**
Refs #26673

### CODE REVIEW

File: doc/ref/modules/index.rst [hunk 1]
```diff
@@ -141,33 +141,34 @@ This will ensure that the text outputter is used.
...
+``__virtual__`` Function
+========================
+
+The ``__virtual__`` function
+returns either a :ref:`string <python2:typesseq>`, :py:data:`True`, or
+:py:data:`False`. If a string is returned then the module is loaded using the
+name of the string as the virtual name. If ``True`` is returned the module is
+loaded using the current module name. If ``False`` is returned the module is not loaded.
+``False`` lets the module perform system checks and prevent loading if dependencies
+are not met.
```
Comment (jfindlay) [inline doc/ref/modules/index.rst]: @jacobhammons, the `__virtual__` function returns either a single argument, the virtualname, `True`, `False` or two arguments, `False`, and a string describing the reason the module was not loaded.

**Comment by rallytime** [comment]
Refs #27724

File: doc/ref/modules/index.rst [hunk 2]
```diff
@@ -141,37 +141,40 @@ This will ensure that the text outputter is used.
...
+``__virtual__`` Function
+========================
+
+The ``__virtual__`` function returns either a :ref:`string <python2:typesseq>`,
+:py:data:`True`, :py:data:`False`, or :py:data:`False` with an :ref:`error
+string <modules-error-info>`. If a string is returned then the module is loaded
+using the name of the string as the virtual name. If ``True`` is returned the
+module is loaded using the current module name. If ``False`` is returned the
+module is not loaded. ``False`` lets the module perform system checks and
+prevent loading if dependencies are not met.
```
Comment (jfindlay) [inline doc/ref/modules/index.rst]: @jacobhammons, you could link to https://docs.saltstack.com/en/latest/ref/modules/index.html#returning-error-information-from-virtual here.

Comment (jacobhammons) [inline doc/ref/modules/index.rst]: line 162 :)

Comment (jfindlay) [inline doc/ref/modules/index.rst]: Nice, thanks.

**Pull request merged**